src/components/screens/editor/editor-screen.tsx

```diff
@@ -182,6 +193,283 @@ const EditorScreen: React.FC<Props> = ({
+              ...selectedIntent.trainingMaterials,
+              [key]: editedTrainingMaterial.id
+            }
+          },
+          storyId: storyId
+        });
+      } catch (error) {
+        throw error;
+      }
+      setStoryData({
+        ...storyData,
+        trainingMaterial: [ ...trainingMaterial, updatedTrainingMaterial ],
+        selectedIntent: updatedIntent,
+        intents: intents.map(item => item.id === updatedIntent.id ? updatedIntent : item)
+      });
+    }
+    setEditingTrainingMaterial(false);
+    setSelectedTrainingMaterialType(null);
+  }
+
+  /**
+   * Event handler for updating knot info
+   * 
+   * @param knot knot with updated info
+   */
+  const onUpdateKnotInfo = async (event: React.ChangeEvent<any>, knot: Knot) => {
+    let { name, value } = event?.target;
+    if (!accessToken || !knots || !knot?.id) {
+      return;
+    }
+
+    const updatedKnot = await Api.getKnotsApi(accessToken).updateKnot({
+      storyId: storyId,
+      knotId: knot.id,
+      knot: { ...knot, [name]: value }
+    });
+
+    setStoryData({
+      ...storyData,
+      knots: knots.map(item => item.id === updatedKnot.id ? updatedKnot : item)
+    });
+  }
+
+  /**
+   * Event handler for updating knot info
+   * 
+   * @param knot knot with updated info
+   */
+  const onUpdateIntentInfo = async (event: React.ChangeEvent<any>) => {
+    const { name, value } = event?.target;
+    if (!accessToken || !intents || !selectedIntent?.id) {
+      return;
+    }
+
+    const updatedIntent = await Api.getIntentsApi(accessToken).updateIntent({
+      storyId: storyId,
+      intentId: selectedIntent.id,
+      intent: { ...selectedIntent, [name]: value }
+    });
+
+    setStoryData({
+      ...storyData,
+      selectedIntent: updatedIntent,
+      intents: intents.map(item => item.id === updatedIntent.id ? updatedIntent : item)
+    });
+  }
+
+  /**
+   * Event handler for update edited training material
+   */
+  const onUpdateEditedTrainingMaterial = (event: React.ChangeEvent<any>) => {
+    const { name, value } = event?.target;
+    if (!editedTrainingMaterial) {
+      return;
+    }
+
+    setEditedTrainingMaterial({ ...editedTrainingMaterial, [name]: value });
+  }
+
+  /**
+   * Converts a training material type into intent training material key
+   * 
+   * @param name training material type key or value
+   * @returns object key
+   */
+  const objectKeyConversion = (name: string): keyof IntentTrainingMaterials => {
+    switch (name) {
+      case TrainingMaterialType.INTENTOPENNLPDOCCAT :
+        return "intentOpenNlpDoccatId";
+      case TrainingMaterialType.INTENTREGEX :
+        return "intentRegexId";
+      case TrainingMaterialType.VARIABLEOPENNLPNER : 
+        return "variableOpenNlpNerId";
+      case TrainingMaterialType.VARIABLEOPENNLPREGEX : 
+        return "variableOpenNlpRegex";
+      default :
+        return "intentOpenNlpDoccatId";
+    }
+  }
```

Comment:
create separate utility class that handles these kinds of tasks. Also can be changed to use object literal, like so:

```suggestion
  const objectKeyConversion = (name: string): keyof IntentTrainingMaterials => ({
    [TrainingMaterialType.INTENTOPENNLPDOCCAT]: "intentOpenNlpDoccatId",
    [TrainingMaterialType.INTENTREGEX]: "intentRegexId",
    [TrainingMaterialType.VARIABLEOPENNLPNER]: "variableOpenNlpNerId",
    [TrainingMaterialType.VARIABLEOPENNLPREGEX]: "variableOpenNlpRegex",
  })[name] ?? "intentOpenNlpDoccatId";
```